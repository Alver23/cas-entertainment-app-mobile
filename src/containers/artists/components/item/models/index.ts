interface IItem {
  id: number;
  firstName: string;
  lastName: string;
  fullName: string;
  active: boolean;
}

export interface IItemProps {
  item: IItem;
  onPress: (id: number) => void;
}
