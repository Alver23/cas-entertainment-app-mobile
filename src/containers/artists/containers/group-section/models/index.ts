export interface IGroupSectionProps {
  groupIds: number[];
}
