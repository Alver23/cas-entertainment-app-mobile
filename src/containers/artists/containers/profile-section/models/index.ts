export interface IProfileSectionProps {
  user: {
    id: number;
    personId: number;
    firstName: string;
    lastName: string;
    active: boolean;
  };
}
