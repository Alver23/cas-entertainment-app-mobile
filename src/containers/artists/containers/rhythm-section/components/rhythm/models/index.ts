// Models
import { ISkill } from '@models/artist/skills';

export interface IRhythmProps {
  skills: ISkill[];
}
