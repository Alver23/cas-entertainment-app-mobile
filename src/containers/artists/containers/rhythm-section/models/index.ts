// Models
import { ISkill } from '@models/artist/skills';

export interface IRhythmSectionProps {
  skills: ISkill[];
}
