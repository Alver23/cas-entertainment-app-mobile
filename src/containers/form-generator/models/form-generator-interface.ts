// Dependencies
import { ReactNode } from 'react';
import { ViewStyle } from 'react-native';
import { ObjectSchema, Shape } from 'yup';

export enum CONTROL_TYPES {
  INPUT = 'input',
}

export type ControlType = CONTROL_TYPES;

export interface ICustomAttributes {
  [prop: string]: string;
}

export interface IField {
  key: string;
  placeholder: string;
  value?: string;
  disabled?: boolean;
  inputStyle?: ViewStyle;
  customAttributes?: ICustomAttributes;
  controlType: ControlType;
}

export interface IFormSchema {
  validations?: ObjectSchema<Shape<object | undefined, Record<string, string>>>;
  fields: IField[];
  onSubmit: () => void;
  initialValues?: Record<string, string | number | boolean>;
  onCancel?: () => void;
  submitLabelButton?: string;
  cancelLabelButton?: string;
  children?: ReactNode;
}
