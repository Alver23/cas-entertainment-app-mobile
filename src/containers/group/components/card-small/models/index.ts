export interface ICardSmallProps {
  title: string;
  description: string;
  state?: string;
}
