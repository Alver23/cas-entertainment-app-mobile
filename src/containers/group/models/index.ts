// Config
import { COMPONENT_TYPES } from '../config';

export interface IGroupListProps {
  groupIds: number[];
  type?: COMPONENT_TYPES;
}
