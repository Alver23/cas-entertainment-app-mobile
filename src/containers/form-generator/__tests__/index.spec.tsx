// Dependencies
import React, { PropsWithChildren } from 'react';
import { Formik } from 'formik';
import { shallow, ShallowWrapper } from 'enzyme';

// Models
import { IField } from '../models/form-generator-interface';

// Containers
import { FormGenerator } from '../index';

// Mocks
import mocks from './mocks.json';

describe('FormGenerator', () => {
  let component: ShallowWrapper<PropsWithChildren<any>>;

  const fields: IField[] = mocks;

  const formikProps = {
    handleSubmit: jest.fn(),
    handleChange: jest.fn(),
    handleBlur: jest.fn(),
    touched: {
      email: true,
    },
    errors: {
      email: 'fake error',
    },
  };

  const props = {
    fields,
    onSubmit: jest.fn(),
  };

  beforeEach(() => {
    component = shallow(<FormGenerator {...props} />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render inputs correctly', () => {
    const { children } = component.find(Formik).props();

    children(formikProps);
  });
});
