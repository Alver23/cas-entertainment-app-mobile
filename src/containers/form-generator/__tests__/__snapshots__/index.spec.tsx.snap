// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGenerator should render correctly 1`] = `
<Formik
  initialValues={
    Object {
      "email": undefined,
    }
  }
  onSubmit={[MockFunction]}
>
  <Component />
</Formik>
`;
